compiler/noirc_frontend/src/parser/parser/enums.rs

```diff
@@ -0,0 +1,281 @@
+use noirc_errors::Span;
+
+use crate::{
+    ast::{Documented, EnumVariant, Ident, ItemVisibility, NoirEnumeration, UnresolvedGenerics},
+    parser::ParserErrorReason,
+    token::{Attribute, SecondaryAttribute, Token},
+};
+
+use super::{
+    parse_many::{separated_by_comma_until_right_brace, separated_by_comma_until_right_paren},
+    Parser,
+};
+
+impl<'a> Parser<'a> {
+    /// Enum = 'enum' identifier Generics '{' EnumVariant* '}'
+    ///
+    /// EnumField = OuterDocComments identifier ':' Type
+    pub(crate) fn parse_enum(
+        &mut self,
+        attributes: Vec<(Attribute, Span)>,
+        visibility: ItemVisibility,
+        start_span: Span,
+    ) -> NoirEnumeration {
+        let attributes = self.validate_secondary_attributes(attributes);
+
+        self.push_error(ParserErrorReason::ExperimentalFeature("Enums"), start_span);
+
+        let Some(name) = self.eat_ident() else {
+            self.expected_identifier();
+            return self.empty_enum(
+                Ident::default(),
+                attributes,
+                visibility,
+                Vec::new(),
+                start_span,
+            );
+        };
+
+        let generics = self.parse_generics();
+
+        if self.eat_semicolons() {
+            return self.empty_enum(name, attributes, visibility, generics, start_span);
+        }
```

Comment:
Should this be removed? This would allow `enum Foo;` which I thought was maybe valid in Rust because of this code but apparently it's not.

Reply:
We could allow it and treat it as the same as `enum Foo {}` but I'll just remove it for consistency.

We could leave it as a separate question whether to support `enum Void {}` for the Void type

Reply:
Sounds good. I then tried `enum Foo {}` in Rust and it worked so I was wondering why Rust then doesn't allow `enum Foo;`. It would be consistent with structs so maybe us supporting that would be good. So up to you to leave it or remove it.